### PR TITLE
feature: web: support displaying and searching package docs.

### PR DESCRIPTION
**Pull request by xlibor**

### CODE REVIEW

File: web/lua/opmserver.lua [hunk 1]
```diff
@@ -884,7 +884,7 @@ do
         end
 
         if not res then
-            ngx.log(ngx.ERR, "fatal response due to query failures")
+            ngx.log(ngx.ERR, "fatal response due to query failures", err)
```
Comment (xiaocang) [inline web/lua/opmserver.lua:887]: Maybe it's better to put a colon here to separate out the error messages.

File: web/lua/opmserver.lua [hunk 2]
```diff
@@ -1795,12 +1807,26 @@ function _M.do_show_package(account, pkg_name)
         pkg_info.dep_info = tab_concat(dep_info, ", ")
     end
 
+    local pkg_doc = pkg_info.doc
+
+    if pkg_doc then
+        local m, err = re_match(pkg_doc, [[<body>(.)</body>]])
```
Comment (xiaocang) [inline web/lua/opmserver.lua:1813]: Is there a typo here?

``` diff
-        local m, err = re_match(pkg_doc, [[<body>(.)</body>]])
+        local m, err = re_match(pkg_doc, [[<body>(.+)</body>]])
```

File: web/lua/opmserver.lua [hunk 3]
```diff
@@ -1795,12 +1807,26 @@ function _M.do_show_package(account, pkg_name)
...
+        if m then
+            pkg_doc = m[1]
+        end
+    end
+
+    if pkg_doc and str_len(pkg_doc) < 10 then
+        pkg_doc = nil
```
Comment (xiaocang) [inline web/lua/opmserver.lua:1820]: Why pkg_doc is set `nil` here?

File: util/opm-pkg-indexer.pl [hunk 4]
```diff
@@ -475,6 +480,20 @@ ()
             }
         }
 
+        my $podfile = "$name-$ver.opm/pod/$name-$ver/$name-$ver.pod";
+        my $quoted_podfile = shell_quote($podfile);
+        my $htmlfile = "/opm/tmp/$final_md5.html";
+        my $quoted_htmlfile = shell_quote($htmlfile);
+
+        warn "pod2html --noindex $quoted_podfile > $quoted_htmlfile";
```
Comment (xiaocang) [inline util/opm-pkg-indexer.pl:488]: This should be used for debugging, right? Maybe it's should be commented.

File: web/lua/opmserver.lua (hunk 2 above)
Comment (xlibor) [reply web/lua/opmserver.lua:1813]: Yes.

File: web/lua/opmserver.lua [hunk 5]
```diff
@@ -1795,12 +1807,25 @@ function _M.do_show_package(account, pkg_name)
         pkg_info.dep_info = tab_concat(dep_info, ", ")
     end
 
+    local pkg_doc = pkg_info.doc
+
+    if pkg_doc then
+        local m, err = re_match(pkg_doc, [[<body>(.+)</body>]])
```
Comment (xiaocang) [inline web/lua/opmserver.lua:1813]: The "jo" option should be added here for optimization of regular matching

**Comment by xiaocang** [comment]
I think we can merge it, cc @doujiang24

File: util/opm-pkg-indexer.pl [hunk 6]
```diff
@@ -475,6 +480,20 @@ ()
             }
         }
 
+        my $podfile = "$name-$ver.opm/pod/$name-$ver/$name-$ver.pod";
+        my $quoted_podfile = shell_quote($podfile);
+        my $htmlfile = "/opm/tmp/$final_md5.html";
+        my $quoted_htmlfile = shell_quote($htmlfile);
```
[the remark this answers is not in the document]
Comment (zhanlangorz) [reply util/opm-pkg-indexer.pl]: 请问新增的这个pod是为了做什么？
代码中没看到是哪里会产生这个pod目录。 
尝试自己搭一个opm的时候， 发现上传的包一直failed, 就是卡在这里， opm build出来的包里根本没有pod目录

**Pull request merged**